improver/nbhood/recursive_filter.py

```diff
@@ -344,47 +345,60 @@ def process(self, cube: Cube, smoothing_coefficients: CubeList) -> Cube:
                 A cubelist containing two cubes of smoothing_coefficient values,
                 one corresponding to smoothing in the x-direction, and the other
                 to smoothing in the y-direction.
+            variable_mask
```

Comment:
I think the original reason to prevent the use of different masks was to enforce that the same mask be applied to different time slices of a cube, e.g applying a consistent land-sea mask, where different masks would be caused by a mistake somewhere.

I've chosen to maintain the default behaviour to be the old behaviour (ie fail if given non-matching masks) for continuity reasons at the moment.